File: src/main/java/nl/chimpgamer/networkmanager/extensions/discordbot/utils/tasks/CreateTokenTask.java
package nl.chimpgamer.networkmanager.extensions.discordbot.utils.tasks;

import lombok.Getter;
import lombok.RequiredArgsConstructor;
import net.dv8tion.jda.core.entities.Message;
import net.dv8tion.jda.core.entities.MessageChannel;
import nl.chimpgamer.networkmanager.extensions.discordbot.DiscordBot;
import nl.chimpgamer.networkmanager.extensions.discordbot.utils.DCMessage;
import nl.chimpgamer.networkmanager.extensions.discordbot.utils.Utils;

@RequiredArgsConstructor
@Getter
public class CreateTokenTask implements Runnable {
    private final DiscordBot discordBot;
    private final MessageChannel channel;
    private final String discordID;

    @Override
    public void run() {
        final String token = Utils.generateToken();
        final Message message = Utils.sendMessageComplete(this.getChannel(), DCMessage.REGISTRATION_TOKEN.getMessage()
                .replace("%newline%", "\n")
                .replace("%token%", token));

        this.getDiscordBot().getDiscordUserManager().insertToken(token, this.getDiscordID(), message);
    }
}

File: shared/src/main/kotlin/nl/chimpgamer/networkmanager/extensions/discordbot/shared/tasks/CreateTokenTask.kt
package nl.chimpgamer.networkmanager.extensions.discordbot.shared.tasks

import net.dv8tion.jda.api.EmbedBuilder
import net.dv8tion.jda.api.events.interaction.command.GenericCommandInteractionEvent
import net.dv8tion.jda.api.utils.data.DataObject
import nl.chimpgamer.networkmanager.extensions.discordbot.shared.DiscordBot
import nl.chimpgamer.networkmanager.extensions.discordbot.shared.configurations.DCMessage
import nl.chimpgamer.networkmanager.extensions.discordbot.shared.utils.Utils

class CreateTokenTask(
    private val discordBot: DiscordBot,
    private val commandInteractionEvent: GenericCommandInteractionEvent
) : Runnable {

    override fun run() {
        val token = Utils.generateToken()
        val msgStr = discordBot.messages.getString(DCMessage.REGISTRATION_TOKEN_RESPONSE)
            .replace("%newline%", "\n")
            .replace("%token%", token)

        if (Utils.isJsonValid(msgStr)) {
            val data = DataObject.fromJson(msgStr)
            val embedBuilder = EmbedBuilder.fromData(data)
            commandInteractionEvent.replyEmbeds(embedBuilder.build()).setEphemeral(true)
                .queue({ success ->
                    discordBot.discordUserManager.insertToken(token, commandInteractionEvent.user.id,
                    success)
                }) { failure ->
                    discordBot.platform.warn("Failed to send REGISTRATION_TOKEN_RESPONSE!")
                    discordBot.platform.warn(failure.localizedMessage)
                }
        } else {
            commandInteractionEvent.reply(msgStr).setEphemeral(true)
                .queue({ success ->
                    discordBot.discordUserManager.insertToken(token, commandInteractionEvent.user.id,
                        success)
                }) { failure ->
                    discordBot.platform.warn("Failed to send REGISTRATION_TOKEN_RESPONSE!")
                    discordBot.platform.warn(failure.localizedMessage)
                }
        }
    }
}